crates/projection/src/horizon.rs

```diff
@@ -0,0 +1,60 @@
+use coordinate_systems::{Camera, Ground, Pixel};
+use linear_algebra::{vector, Isometry3, Point2, Vector2, Vector3};
+use serde::{Deserialize, Serialize};
+use serialize_hierarchy::SerializeHierarchy;
+
+use crate::intrinsic::Intrinsic;
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize, SerializeHierarchy)]
+pub struct Horizon {
+    pub vanishing_point: Point2<Pixel>,
+    pub normal: Vector2<Pixel>,
+}
+
+impl Horizon {
+    pub fn horizon_y_minimum(&self) -> f32 {
+        self.y_at_x(0.0).min(self.y_at_x(640.0))
+    }
+
+    pub fn y_at_x(&self, x: f32) -> f32 {
+        -self.normal.x() * (x - self.vanishing_point.x()) / self.normal.y()
+            + self.vanishing_point.y()
+    }
```

Comment:
You divide by `normal.y` here. This will break for with the derived Default implementation which creates a 0 normal vector, which is supposed to be a unit vector.